### ISSUE TITLE
`easy_install tensorflow-gpu` fails

### ISSUE THREAD

**Issue by sjperkins** [tensorflow/tensorflow#6540]
A pip install of tensorflow-gpu and tensorflow works great.

```bash
$ pip install tensorflow-gpu
```

However, when one is trying to create a package with a dependency on tensorflow, one usually has to do the following in `setup.py`:

```python
from setuptools import setup

setup(name='mypkg',
    setup_requires=["tensorflow-gpu == 0.12.0", ...],
    install_requires=["tensorflow-gpu == 0.12.0", ...],
)
```

Unfortunately, using the above as dependencies fail because setuptools defers to `easy_install` rather than `pip`.

```python
Searching for tensorflow-gpu
Reading https://pypi.python.org/simple/tensorflow-gpu/
No local packages or working download links found for tensorflow-gpu
error: Could not find suitable distribution for Requirement.parse('tensorflow-gpu')
```

easy_install doesn't seem to be able to recognise the wheels found at https://pypi.python.org/simple/tensorflow-gpu/

### Environment info
Operating System: Ubuntu 16.04

Installed version of CUDA and cuDNN: 
(please attach the output of `ls -l /path/to/cuda/lib/libcud*`):

```bash
$ ls -l /usr/local/cuda/lib64/libcud*
-rw-r--r-- 1 root root 558720 Sep 15 01:02 /usr/local/cuda/lib64/libcudadevrt.a
lrwxrwxrwx 1 root root     16 Sep 15 01:05 /usr/local/cuda/lib64/libcudart.so -> libcudart.so.8.0
lrwxrwxrwx 1 root root     19 Sep 15 01:05 /usr/local/cuda/lib64/libcudart.so.8.0 -> libcudart.so.8.0.44
-rw-r--r-- 1 root root 415432 Sep 15 01:02 /usr/local/cuda/lib64/libcudart.so.8.0.44
-rw-r--r-- 1 root root 775162 Sep 15 01:02 /usr/local/cuda/lib64/libcudart_static.a

```

If installed from binary pip package, provide:

1. A link to the pip package you installed: N/A
2. The output from `python -c "import tensorflow; print(tensorflow.__version__)"`. 0.12.0

If installed from source, provide 

1. The commit hash (`git rev-parse HEAD`): N/A
2. The output of `bazel version`: N/A

### If possible, provide a minimal reproducible example (We usually don't have time to read hundreds of lines of your code)

```bash
$ easy_install tensorflow-gpu
Searching for tensorflow-gpu
Reading https://pypi.python.org/simple/tensorflow-gpu/
No local packages or working download links found for tensorflow-gpu
error: Could not find suitable distribution for Requirement.parse('tensorflow-gpu')
```

But inspection of https://pypi.python.org/simple/tensorflow-gpu/ shows:

```
Links for tensorflow-gpu
tensorflow_gpu-0.12.0-cp34-cp34m-manylinux1_x86_64.whl
tensorflow_gpu-0.12.0-cp35-cp35m-manylinux1_x86_64.whl
tensorflow_gpu-0.12.0rc1-cp35-cp35m-manylinux1_x86_64.whl
tensorflow_gpu-0.12.0rc1-cp35-cp35m-win_amd64.whl
tensorflow_gpu-0.12.0rc1-cp34-cp34m-manylinux1_x86_64.whl
tensorflow_gpu-0.12.0rc1-cp27-cp27m-macosx_10_11_intel.whl
tensorflow_gpu-0.12.0-cp35-cp35m-win_amd64.whl
tensorflow_gpu-0.12.0-cp27-cp27mu-manylinux1_x86_64.whl
tensorflow_gpu-0.12.0rc1-cp35-cp35m-macosx_10_11_x86_64.whl
tensorflow_gpu-0.12.0-cp27-cp27m-macosx_10_11_intel.whl
tensorflow_gpu-0.12.0rc1-cp27-cp27mu-manylinux1_x86_64.whl
tensorflow_gpu-0.12.0-cp35-cp35m-macosx_10_11_x86_64.whl
tensorflow_gpu-0.12.0rc0-cp27-cp27mu-manylinux1_x86_64.whl
tensorflow_gpu-0.12.0rc0-cp35-cp35m-win_amd64.whl
```

### Logs or other output that would be helpful
(If logs are large, please upload as attachment or provide link).

**Comment by gunan** [comment]
@yifeif Is this because we are hosting tensorflow gpu pip package at the new pypi backend?
Or maybe easy_install uses some workflow similar to old pip versions?

@sjperkins for context, tensorflow pip packages exceed the old pypi package size limit.
Therefore, they have to be hosted in their new datastore. This makes our packages uninstallable using old versions of pip. So maybe this is an issue with the way easy_install tries to fetch the package?

**Comment by yifeif** [comment]
It could be something with the '_' and '-' replacement. https://pypi.python.org/simple/tensorflow_gpu/  seems to work as well. @sjperkins do you mind try replacing tensorflow-gpu with tensorflow_gpu?

**Comment by sjperkins** [comment]
Sorry should have mentioned, underscore also does not work.

**Comment by yifeif** [comment]
According to [this page](https://packaging.python.org/pip_easy_install/), easy_install does not install from wheels, and it only installs from egg. We currently only build wheel packages. Looks like setup_requires section of setup.py is always fulfilled by easy_install http://pip.readthedocs.io/en/latest/reference/pip_install/#controlling-setup-requires. Do you need tensorflow in both setup_requires and install_requires, or would install_requires be sufficient?

FYI @martinwicke

**Comment by sjperkins** [comment]
**tl;dr** Just use `pip` for everything, rather than `python setup.py ...` (a.k.a. `easy_install`) which is deprecated

Feel free to close if you don't want to support the legacy `easy_install` stuff.

@yifeif My use case for placing `tensorflow-gpu` in `setup_requires` is compiling custom ops into a shared library (python extension) during the installation process of a module I'm building. So I wanted  tensorflow to be installed after the setup.py is run but before the extension can be compiled.

For others encountering this issue, just use pip for everything. I was using  `python setup.py ...` calls which invoke easy_install (see below), but this is [legacy](https://docs.python.org/2/install/) behaviour and [pip](https://docs.python.org/2/installing/index.html) is now the way forward. So this issue arose out of different module install methods. Not sure if google wants to support the old `easy_install` style....

I made a small module [(testsetup.zip)](https://github.com/tensorflow/tensorflow/files/677030/testsetup.zip)  to test things out for future reference:

```python
import ez_setup
ez_setup.use_setuptools()

from setuptools import setup, find_packages
from setuptools.command.install import install
from setuptools.command.develop import develop

class DevelopCommand(develop):
    def run(self):
        #import tensorflow
        develop.run(self)

class InstallCommand(install):
    def run(self):
        #import tensorflow
        install.run(self)

setup(name="testsetup",
    #setup_requires=['tensorflow-gpu'],
    install_requires=['tensorflow-gpu'],
    packages=find_packages(),
    cmdclass={
        'install' : InstallCommand,
        'develop' : DevelopCommand,
     },
)

```

Within the install directory if one calls:

- `python setup.py install` it succeeds without installing dependencies!
- `python setup.py develop` it defers to easy_install and fails because `easy_install` can't find an egg.
- `pip install .` it succeeds if setup_requires is commented out, but fails otherwise. However, `setup_requires` isn't strictly required since `install_requires` will be run before it and the tensorflow dependency will be inplace before the bulk of the setup() code is run. Its certainly installed before InstallCommand.run() is called.
- `pip install -e .` it succeeds if setup_requires is commented out, but fails otherwise.

**Comment by gunan** [comment]
Building and distributing an `egg` file is not planned for tensorflow, as it is the old deprecated standard.
I will close this issue now.
